lib/liquid/context.rb

```diff
@@ -168,17 +195,8 @@ def find_variable(key, raise_on_not_found: true)
 
       variable = nil
 
-      if scope.nil?
-        @environments.each do |e|
-          variable = lookup_and_evaluate(e, key, raise_on_not_found: raise_on_not_found)
-          # When lookup returned a value OR there is no value but the lookup also did not raise
-          # then it is the value we are looking for.
-          if !variable.nil? || @strict_variables && raise_on_not_found
-            scope = e
-            break
-          end
-        end
-      end
+      scope ||= find_environment(@environments, key, raise_on_not_found: raise_on_not_found)
```

Comment:
It seems we're doing the `lookup_and_evaluate` in `find_environment` but then dismissing the value, and doing it again on L202 below. Could we optimize this?